app/models/manageiq/providers/redfish/physical_infra_manager/operations/power.rb

```diff
@@ -0,0 +1,65 @@
+module ManageIQ::Providers::Redfish
+  module PhysicalInfraManager::Operations::Power
+    # Keep this in sync with app/models/physical_server/operations/power.rb in
+    # core and ResetType enum in Redfish Resource type. Name of the method
+    # comes from the core and the action name used in the reset call from the
+    # ResetType enum.
+    #
+    # NOTE: Not all reset operations are implemented on all servers, so any of
+    # the methods listed here can fail. We need to find a way to let those
+    # failures bubble up to the user interface somehow or risk having a
+    # completely useless tool.
+
+    def power_on(server, options = {})
+      reset_server("On", server, options)
+    end
+
+    def power_off(server, options = {})
+      reset_server("GracefulShutdown", server, options)
+    end
+
+    def power_off_now(server, options = {})
+      reset_server("ForceOff", server, options)
+    end
+
+    def restart(server, options = {})
+      reset_server("GracefulRestart", server, options)
+    end
+
+    def restart_now(server, options = {})
+      reset_server("ForceRestart", server, options)
+    end
+
+    def restart_to_sys_setup(_args, _options = {})
+      $redfish_log.error("Restarting to system setup is not supported.")
+    end
+
+    def restart_mgmt_controller(_server, _options = {})
+      # TODO(tadeboro): This operation is not well defined, since server can
+      # (and usually is) managed by more that one manager.
+      $redfish_log.error("Restarting management controller is not supported.")
+    end
+
+    private
+
+    def reset_server(reset_type, server, options = {})
+      $redfish_log.info("Requesting #{reset_type} for #{server.ems_ref}.")
+      with_provider_connection(options) do |client|
+        system = client.find(server.ems_ref)
+        if system.nil?
+          $redfish_log.error("#{server.ems_ref} does not exist anymore.")
```

Comment:
We have to understand if log entry is sufficient, because this will again not get the correct into to the caller - it just silently failed.

Reply:
Again, see other comments on why things are the way they are.